core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -255,7 +255,9 @@ private[spark] class BlockManager(
   // specified memory threshold. Files will be deleted automatically based on weak reference.
   // Exposed for test
   private[storage] val remoteBlockTempFileManager =
-    new BlockManager.RemoteBlockDownloadFileManager(this)
+    new BlockManager.RemoteBlockDownloadFileManager(
```

Comment:
I wasn't clear why this had to change?

Reply:
The test was failing with `NullPointerException` cause the `SparkEnv.get` wasn't initialized.
Considering that the `RemoteBlockDownloadFileManager` accepts an instance of `BlockManager` which already has `SecurityManager` it seemed logical to just pass the `encryptedKey` too so it didn't rely on the static env in the lazy val.

As I understand the alternative would be to construct a mock environment and call the static `SparkEnv.set` and I see that this approach is taken in other places, but I do not understand how it works reliably without removing it after use - it's just left there. Hence I decided to remove the reliance on the static env so it's easier to test.